### PR TITLE
[Feature] Do not allow beta to override stable repository if both active

### PR DESCRIPTION
**Pull request by GermanBluefox**
- Do not allow to be active together the stable and beta repositories.
- Allow addressing the repositories by index in the table

To avoid such a situation
image

**Tests**
- [ ] I have added tests to avoid a recursion of this bug
- [x] It is not possible to test for this bug

Only CLI files changed

**Documentation**
- [x] I have documented the new feature

### CODE REVIEW

**Review by Apollon77 — CHANGES_REQUESTED** [review]
What exactly is the point of working with index? This adds a lot of edge cases as you can see in the code you needed to add. In fact you can not assume any order anyway, so what exactly is the point here?

if it gets easier we could also not use a joined parameter "name/index" but use different comands that use index which makes it easier too

**Review by foxriver76 — APPROVED** [review]

File: packages/cli/src/lib/setup/setupRepo.ts [hunk 1]
```diff
@@ -421,25 +427,46 @@ export class Repo {
...
+            const confObj = await this.objects.getObject('system.config');
             if (typeof confObj?.common.activeRepo === 'string') {
                 confObj.common.activeRepo = [confObj.common.activeRepo];
             }
 
             if (confObj && !confObj.common.activeRepo.includes(repoName)) {
                 confObj.common.activeRepo.push(repoName);
+
+                // Check that stable and beta repositories cannot be active at the same time.
+                if (repoName.toLowerCase().includes('beta')) {
```
[the remark this answers is not in the document]
Comment (GermanBluefox) [reply packages/cli/src/lib/setup/setupRepo.ts]: Changed to startsWith

**Review by foxriver76 — APPROVED** [review]

**Comment by GermanBluefox** [comment]
I still do not understand, If we have index, why not to use it instead of writing "my repo" ?
Same logic is used for backends

**Comment by foxriver76** [comment]
> I still do not understand, If we have index, why not to use it instead of writing "my repo" ? Same logic is used for restoring

Hm index is now supported? but of course not for `add` or `addset` just for `set`and `del`, as the index will just be the next free and you cannot specify like 100 out of nowhere. Or what is the problem now?

**Review by Apollon77 — APPROVED** [review]

**Pull request merged**